config/base/src/lib.rs

```diff
@@ -256,3 +235,103 @@ pub mod view {
         pub const IS_HAS_VIEW: bool = true;
     }
 }
+
+pub mod proxy {
+    //! Module for configuration proxies' traits
+
+    use super::*;
+
+    /// Trait for dynamic and asynchronous configuration via maintenance endpoint for Rust structures
+    pub trait Configuration: Serialize + DeserializeOwned {
+        /// Error type returned by methods of this trait
+        type Error;
+
+        /// Return the JSON value of a given field
+        /// # Errors
+        /// Fails if field was unknown
+        fn get(&self, field: &'_ str) -> Result<Value, Self::Error> {
+            self.get_recursive([field])
+        }
+
+        /// Return the JSON value of a given inner field of arbitrary inner depth
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_recursive<'tl, T>(&self, inner_field: T) -> Result<Value, Self::Error>
+        where
+            T: AsRef<[&'tl str]> + Send + 'tl;
+
+        /// Get documentation of a given inner field of arbitrary depth
+        ///
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_doc_recursive<'tl>(
+            field: impl AsRef<[&'tl str]>,
+        ) -> Result<Option<String>, Self::Error>;
+
+        /// Get documentation of a given field
+        /// # Errors
+        /// Fails if field was unknown
+        fn get_doc(field: &str) -> Result<Option<String>, Self::Error> {
+            Self::get_doc_recursive([field])
+        }
+
+        /// Return documentation for all fields in a form of a JSON object
+        fn get_docs() -> Value;
+
+        /// Get inner documentation for non-leaf fields
+        fn get_inner_docs() -> String;
+    }
+
+    /// Trait for configuration loading and deserialization
+    pub trait Configurable:
```

Comment:
Followed the naming suggested by @appetrosyan. Can agree that it does feel a bit cumbersome, but also see that `Configurable` + `Configuration` convey the purpose of separate traits more clearly, as `Proxy` could be a bit misleading.

Reply:
Don't like the name. It would have been a good name if the traits were a bit larger. In this current implementation I would prefer `FromPartial` or `Combine`

Reply:
Opted for `Combine` + `Documented` combo